src/parseWithPointers.ts

```diff
@@ -12,30 +22,408 @@ export const parseWithPointers: SourceMapParser = <T>(value: string): IParserRes
+  const onArrayEnd = toNoArgVisit(visitor.onArrayEnd);
+  const onLiteralValue = toOneArgVisit(visitor.onLiteralValue);
+  const onSeparator = toOneArgVisit(visitor.onSeparator);
+  const onComment = toNoArgVisit(visitor.onComment);
+  const onError = toOneArgVisit(visitor.onError);
+  const onNewLine = toOneArgVisit(visitor.onNewLine);
+
+  const disallowComments = options && options.disallowComments;
+  const allowTrailingComma = options && options.allowTrailingComma;
+  function scanNext(): SyntaxKind {
+    while (true) {
+      const token = _scanner.scan();
+      switch (_scanner.getTokenError()) {
+        case ScanError.InvalidUnicode:
+          handleError(ParseErrorCode.InvalidUnicode);
+          break;
+        case ScanError.InvalidEscapeCharacter:
+          handleError(ParseErrorCode.InvalidEscapeCharacter);
+          break;
+        case ScanError.UnexpectedEndOfNumber:
+          handleError(ParseErrorCode.UnexpectedEndOfNumber);
+          break;
+        case ScanError.UnexpectedEndOfComment:
+          if (!disallowComments) {
+            handleError(ParseErrorCode.UnexpectedEndOfComment);
+          }
+          break;
+        case ScanError.UnexpectedEndOfString:
+          handleError(ParseErrorCode.UnexpectedEndOfString);
+          break;
+        case ScanError.InvalidCharacter:
+          handleError(ParseErrorCode.InvalidCharacter);
+          break;
+      }
+      switch (token) {
+        case SyntaxKind.LineCommentTrivia:
+        case SyntaxKind.BlockCommentTrivia:
+          if (disallowComments) {
+            handleError(ParseErrorCode.InvalidCommentToken);
+          } else {
+            onComment();
+          }
+          break;
+        case SyntaxKind.Unknown:
+          handleError(ParseErrorCode.InvalidSymbol);
+          break;
+        case SyntaxKind.Trivia:
+          break;
+        case SyntaxKind.LineBreakTrivia:
+          onNewLine(++lineNumber);
+          break;
+        default:
+          return token;
+      }
+    }
+  }
+
+  function handleError(error: ParseErrorCode, skipUntilAfter: SyntaxKind[] = [], skipUntil: SyntaxKind[] = []): void {
+    onError(error);
+    if (skipUntilAfter.length + skipUntil.length > 0) {
+      let token = _scanner.getToken();
+      while (token !== SyntaxKind.EOF) {
+        if (skipUntilAfter.indexOf(token) !== -1) {
+          scanNext();
+          break;
+        } else if (skipUntil.indexOf(token) !== -1) {
+          break;
+        }
+        token = scanNext();
+      }
+    }
+  }
+
+  function parseString(isValue: boolean): boolean {
+    const value = _scanner.getTokenValue();
+    if (isValue) {
+      onLiteralValue(value);
+    } else {
+      onObjectProperty(value);
+    }
+    scanNext();
+    return true;
+  }
+
+  function parseLiteral(): boolean {
+    switch (_scanner.getToken()) {
+      case SyntaxKind.NumericLiteral:
+        let value = 0;
+        try {
+          value = JSON.parse(_scanner.getTokenValue());
```

Comment:
Shouldn't this be a `parseFloat` call?

```suggestion
          value = parseFloat(_scanner.getTokenValue());
```

Reply:
Ha, good question.
Why would you put `parseFloat` here?
Not saying you are wrong, just want to know the reason. I don't really know what the actual difference is, would need to check out the spec.

Reply:
If I understand the code correctly, the value we're getting should be a number. If it's not, we should get an error. If (for some reason, such as an internal bug) the object is a string or something else, JSON.parse would return something anyway (which we do not want).

Reply:
> if (for some reason, such as an internal bug) the object is a string or something else

The above shouldn't happen since the token we operate on must be number. If it's not a number, it's likely to be a more serious issue (like nothing works as expected)

Reply:
Your call, I'd still change it to `parseFloat`, but if you feel confident I'm ok with that

Reply:
@XVincentX 
Take a look at the condition liner after.
```
  if (typeof value !== 'number') {
            handleError(ParseErrorCode.InvalidNumberFormat);
            value = 0;
          }
```
We do handle it.